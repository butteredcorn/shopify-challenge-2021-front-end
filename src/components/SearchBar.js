import styled from 'styled-components'
import '../styles/SearchBar.css'

const SearchBarContainer = styled.section`
    background-color: ${props => props.theme.bg.primary};
    color: ${props => props.theme.text.tertiary};
    border: 1px solid ${props => props.theme.bg.primary};
`

const SearchBarInner = styled.section`
    background-color: ${props => props.theme.bg.secondary};
    color: ${props => props.theme.text.tertiary};
    border: 1px solid ${props => props.theme.bg.primary};
`

const Search = styled.span`
    border: 1px solid black;
`

const SearchBarTitle = styled.label`
    color: ${props => props.theme.header.primary};
`

const SearchBar = ({input, setInput, handleEvent}) => {
    return (
        <SearchBarContainer className="search-bar-container">
            <SearchBarInner className="search-bar-inner">
                <SearchBarTitle className="search-bar-title">Movie title</SearchBarTitle>
                <Search className="search"><img className="search-icon" src={"https://cdn4.iconfinder.com/data/icons/glyphs/24/icons_search-64.png"} alt="search icon" onClick={handleEvent}/><input className="search-input" type="text" name="keyword" value={input} onChange={setInput} onKeyPress={handleEvent}/></Search>
            </SearchBarInner>
        </SearchBarContainer>
    );
  };
  export default SearchBar;