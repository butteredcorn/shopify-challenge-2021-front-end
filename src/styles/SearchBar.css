.search-bar-container {
    height: 10vh;
    /* margin: 1rem; */
}

.search-bar-inner {
    display: flex;
    flex-direction: column;
    padding: 1rem;
    margin: 1rem;
}

.search-bar-title {
    font-size: 1.5rem;
}

.search-input {
    width: 95%;
}