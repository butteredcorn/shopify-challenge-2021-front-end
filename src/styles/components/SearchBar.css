.search-bar-container {
    min-height: 10vh;
}

.search-bar-inner {
    display: flex;
    flex-direction: column;
    padding: 1rem;
    margin: 1rem;
}

.search-bar-title {
    font-size: 1.5rem;
    font-weight: bold;
}

.search {
    display: flex;
    flex-direction: row;
    margin: 0;
    padding: 0;
}

.search-icon {
    height: 28px;
    padding: 0.2rem;
}

.search-input {
    width: 100%;
    padding-right: 0.5rem;
    padding-left: 0.5rem;
    border: none;
    font-size: 1rem;
}

@media only screen and (max-width: 600px) {

    .search-bar-inner {
        margin-top: 0;
    }
}