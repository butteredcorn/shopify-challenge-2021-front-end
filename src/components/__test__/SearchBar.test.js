import { render, screen, cleanup } from '@testing-library/react';
import ThemesProvider from '../../providers/ThemeProvider'
import renderer from 'react-test-renderer'
import SearchBar from '../SearchBar';

afterEach(cleanup);

it('renders SearchBar component with title "Movie title"', () => {
  render(<ThemesProvider><SearchBar/></ThemesProvider>);
  const message = screen.getByText(/Movie title/i);
  expect(message).toBeInTheDocument();
});

it('matches a snapshot', () => {
  const json = renderer.create(<ThemesProvider><SearchBar/></ThemesProvider>).toJSON();
  expect(json).toMatchSnapshot();
});