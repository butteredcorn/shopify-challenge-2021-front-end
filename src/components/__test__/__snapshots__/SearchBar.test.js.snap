// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<section
  className="sc-bdvvaa ktWoHH search-bar-container"
>
  <section
    className="sc-gsDJrp eZjVwY search-bar-inner"
  >
    <label
      className="sc-hKwCoD eqUIOT search-bar-title"
    >
      Movie title
    </label>
    <span
      className="sc-dkPtyc hboLrn search"
    >
      <img
        alt="search icon"
        className="search-icon"
        src="https://cdn4.iconfinder.com/data/icons/glyphs/24/icons_search-64.png"
      />
      <input
        className="search-input"
        name="keyword"
        type="text"
      />
    </span>
  </section>
</section>
`;
